app/models/host.rb

```diff
@@ -1936,17 +1936,15 @@ def self.node_types
   end
 
   def self.count_of_openstack_hosts
-    ems = ManageIQ::Providers::Openstack::InfraManager.pluck(:id)
-    Host.where(:ems_id => ems).count
+    ManageIQ::Providers::Openstack::InfraManager::Host.count
   end
 
   def self.count_of_non_openstack_hosts
-    ems = ManageIQ::Providers::Openstack::InfraManager.pluck(:id)
-    Host.where(Host.arel_table[:ems_id].not_in(ems)).count
+    Host.where.not(:type => ManageIQ::Providers::Openstack::InfraManager::Host).count
```

Comment:
Will this handle subclasses on `ManageIQ::Providers::Openstack::InfraManager::Host`?  Not sure it matters, but to be safe.